pyk/src/pyk/proof/reachability.py

```diff
@@ -584,7 +584,9 @@ def read_proof_data(proof_dir: Path, id: str) -> APRProof:
             kcfg._resolve(int(node_id)): proof_id for node_id, proof_id in proof_dict['node_refutations'].items()
         }
 
-        prior_loops_cache = {int(k): v for k, v in proof_dict.get('loops_cache', {}).items()}
+        prior_loops_cache: dict[int, tuple[int, ...]] = {
+            int(k): tuple(v) for k, v in proof_dict.get('loops_cache', {}).items()
+        }
```

Comment:
Condiser pushing the conversion down into `APRPRoof.__init__`: https://github.com/runtimeverification/k/blob/08a32dea92f62c112b9fc4ae88b7efc45ead38aa/pyk/src/pyk/proof/reachability.py#L151